src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -123,6 +142,69 @@ def ev_charger_pool(
             )
         return self._ev_charger_pools[key]
 
+    def battery_pool(
+        self,
+        battery_ids: abc.Set[int] | None = None,
+    ) -> BatteryPool:
+        """Return the corresponding BatteryPool instance for the given ids.
+
+        If an BatteryPool instance for the given ids doesn't exist, a new one is
+        created and returned.
+
+        Args:
+            battery_ids: Optional set of IDs of batteries to be managed by the
+                BatteryPool.
+
+        Returns:
+            An BatteryPool instance.
+        """
+        from ..timeseries.battery_pool import BatteryPool
+
+        if not self._power_distributing_actor:
+            self._start_power_distributing_actor()
+
+        # We use frozenset to make a hashable key from the input set.
+        key: frozenset[int] = frozenset()
+        if battery_ids is not None:
+            key = frozenset(battery_ids)
+
+        if key not in self._battery_pools:
+            self._battery_pools[key] = BatteryPool(
+                batteries_status_receiver=self._battery_status_channel.new_receiver(),
+                min_update_interval=timedelta(
+                    seconds=self._resampler_config.resampling_period_s
+                ),
+                batteries_id=battery_ids,
+            )
+
+        return self._battery_pools[key]
+
+    def power_distributor_handle(self) -> Bidirectional.Handle[Request, Result]:
```

Comment:
should this function  be named `power_distributing_handle` instead to be aligned with other function names or the actor name?